components/common/src/command/package/install.rs

```diff
@@ -702,11 +704,38 @@ impl<'a> InstallTask<'a> {
     where
         T: UIWriter,
     {
-        artifact.unpack(Some(self.fs_root_path))?;
-        ui.status(Status::Installed, artifact.ident()?)?;
+        let ident = &artifact.ident()?;
+        let real_install_path = &pkg_install_path(ident, Some(self.fs_root_path));
+        let real_install_base = Path::new(real_install_path)
+            .parent()
+            .ok_or_else(|| Error::PackageUnpackFailed("Could not determine parent directory for package unpack location".to_owned()))?;
+        let temp_dir = self.temporary_unpack_directory(real_install_base, real_install_path)?;
+        let temp_install_path = &pkg_install_path(ident, Some(temp_dir.path()));
+        artifact.unpack(Some(temp_dir.path()))?;
+        fs::rename(temp_install_path, real_install_path)?;
+        self.fsync_dir(real_install_base)?;
```

Comment:
Since the comment isn't necessary, in my opinion, I think you can remove this new function and just have this inline:
```rust
fs::File::open(real_install_base).and_then(|f| f.sync_all())?;
```